src/lib/comms/sol-coap.c

```diff
@@ -900,6 +910,8 @@ sol_coap_send_packet_with_reply(struct sol_coap_server *server, struct sol_coap_
     }
 
     sol_coap_header_get_id(pkt, &reply->id);
+    reply->reentrant.in_use = false;
+    reply->reentrant.delete_me = false;
```

Comment:
do you think this would be useful:

```c
static inline void
sol_reentrant_init(struct sol_reentrant *r)
{
    r->in_use = false;
    r->delete_me = false;
}

...
    sol_reentrant_init(&reply->reentrant);
```